File: runme/mcgen/local/datan/exasrc/e4mtx.c
#include <stdio.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__3 = 3, c__5 = 5, c__6 = 6;

main(void)
{
    /* Initialized data */

    static double v[6] = { 1.,2.,0.,4.,3.,4. },
                  c[6] = { 1.,2.,0.,4.,3.,4. }, b, up;

/* identify program to user */
    printf("%s\n", "Program E4MTX demonstrates use of");
    printf("%s\n", "MTXHSD and MTXHST");
    printf("%s\n\n", "for Householder transformations.");
    printf("%s\n", "Output produced by program E4MTX.");
    printf("%s\n", "------------------------------------------");
/* write input data */
    printf("%s%2li%s%2li%s%2li\n", "N = ", c__6, ", LP = ", c__3,
           ", L = ", c__5);
/* write initial vector */
    printf("%s\n", "V =");
    mtxwrt_(v, &c__1, &c__6);
    printf("%s\n\n", "------------------------------------------");
/* demonstrate MTXHSD */
    mtxhsd_(v, &up, &b, &c__6, &c__3, &c__5);
    printf("%s\n", "CALL MTXHSD(U,UP,B,N,LP,L) yields ");
    printf("%s%10.5lf%s%10.5lf\n\n", "UP = ", up, ", B = ", b);
/* demonstrate MTXHST */
    mtxhst_(v, &up, &b, c, &c__6, &c__3, &c__5);
    printf("%s\n", "With the above values of V, UP and B");
    printf("%s\n", "CALL MTXHST(V,UP,B,C,N,LP,L) yields ");
    printf("%s\n", "C =");
    mtxwrt_(c, &c__1, &c__6);
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e4mtx.f
      PROGRAM E4MTX
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER (N=6,L=5,LP=3)
      DIMENSION V(N),C(N)
      DATA V/1.D0,2.D0,0.D0,4.D0,3.D0,4.D0/
      DATA C/1.D0,2.D0,0.D0,4.D0,3.D0,4.D0/
C identify program to user
      WRITE(*,*)' Program E4MTX demonstrates use of'
      WRITE(*,*)' MTXHSD and MTXHST'
      WRITE(*,*)' for Householder transformations'
      WRITE(*,*)' '
      WRITE(*,*)' Output produced by program E4MTX'
      WRITE(*,'(A)')' ------------------------------------------'
C write input data
      WRITE(*,'(A,I2,A,I2,A,I2)')' N = ',N,', LP = ',LP,', L = ',L
C write initial vector
      WRITE(*,*)' V ='
      CALL MTXWRT(V,1,N)
      WRITE(*,'(A)')' ------------------------------------------'
      WRITE(*,*)' '
C demonstrate MTXHSD
      CALL MTXHSD(V,UP,B,N,LP,L)
      WRITE(*,'(A)')' CALL MTXHSD(U,UP,B,N,LP,L) yields '
      WRITE(*,'(A,F10.5,A,F10.5)')' UP = ',UP,', B = ',B
      WRITE(*,*)' '
C demonstrate MTXHST
      CALL MTXHST(V,UP,B,C,N,LP,L)
      WRITE(*,*)' With the above values of V, UP and B'
      WRITE(*,'(A)')' CALL MTXHST(V,UP,B,C,N,LP,L) yields '
      WRITE(*,*)' C ='
      CALL MTXWRT(C,1,N)
      END
